### PR TITLE
Displays devfile v2 getting started if devWorkspace toggle is enabled

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Display in getting started the `v2` links of devfile registries if DevWorkspace support is enabled.
image

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19923

### CODE REVIEW

**Comment by benoitf** [comment]
@olexii4 @akurinnoy if you could tell me if it's using a good approach, what I need to change, follow, etc. (For sure I know that it may require tests but if the global idea is totally wrong then :D )

**Comment by dmytro-ndp** [comment]
[dashboard-ci-test]

File: src/store/DevfileRegistries/selectors.ts [hunk 1]
```diff
@@ -14,22 +14,29 @@ import { createSelector } from 'reselect';
 import { AppState } from '../';
 import match from '../../services/helpers/filter';
 
-const selectState = (state: AppState) => state.devfileRegistries;
+const selectState = (state: AppState) => ({ devfileRegistries: state.devfileRegistries, workspacesSettings: state.workspacesSettings });
```
Comment (akurinnoy) [inline src/store/DevfileRegistries/selectors.ts:17]: I think of importing the `selectWorkspacesSettingsState` or just creating this state so you don't need to modify other selectors but only `selectRegistriesMetadata`:

```typescript

import { selectWorkspacesSettingsState } from '../Workspaces/Settings/selectors';
// or
const selectWorkspacesSettingsState = (state: AppState) => state.workspacesSettings;

const selectState = (state: AppState) => state.devfileRegistries;

export const selectRegistriesMetadata = createSelector(
  selectState,
  selectWorkspacesSettingsState,
  (devfileRegistriesState, workspacesSettingsState) => {
    const registriesMetadata = Object.values(devfileRegistriesState.registries)
      .map(registryMetadata => registryMetadata.metadata || []);
    const metadata = mergeRegistriesMetadata(registriesMetadata);
    const cheDevworkspaceEnabled = workspacesSettingsState.settings['che.devworkspaces.enabled'] === 'true';
    if (cheDevworkspaceEnabled) {
      return filterDevfileV2Metadata(metadata);
    } else {
      return metadata;
    }

  }
);
```

Comment (benoitf) [reply src/store/DevfileRegistries/selectors.ts:17]: yes it looks far better

**Comment by benoitf** [comment]
closing in favor of https://github.com/eclipse-che/che-dashboard/pull/262 as I should push to repository instead of a fork to have all PR checks green

**Pull request closed without merging**